cargo-dist/src/migrate/from_v0.rs

```diff
@@ -1,5 +1,60 @@
-use crate::errors::DistResult;
+use crate::config::v0::V0DistConfig;
+use crate::config::v1::DistConfig;
+use crate::{config, errors::DistResult};
+use axoasset::toml;
+
+// A purely-functional subset of `do_migrate_from_v0()`:
+// takes a DistMetadata (v0) and returns an equivalent DistConfig (v1)
+// without touching files on disk.
+//
+// This could theoretically have unit tests written for it, but we would
+// need to make every piece of DistConfig derive/impl PartialEq, which is
+// a lot. -duckinator
+fn migrate_from_v0_dist_config(old_config: V0DistConfig) -> DistConfig {
+    let V0DistConfig {
+        dist,
+        workspace,
+        package,
+    } = old_config;
+
+    let dist = dist.map(|dist| {
+        dist.to_toml_layer(true)
+            .with_current_config_version()
+            .with_current_dist_version()
+    });
+
+    config::v1::DistConfig {
+        dist,
+        workspace,
+        package,
+    }
+}
```

Comment:
I tried adding unit tests for this, but it would cause a ton of code churn unless I figure out a better approach.

I found it easier to reason through the code after splitting it up, though, so I left it like this.